UefiPayloadPkg/PciPlatformDxe/PciPlatformDxe.c

```diff
@@ -43,6 +45,88 @@ PciPlatformPrepController(
   return EFI_UNSUPPORTED;
 }
 
+STATIC
+BOOLEAN
+IsVgaDevice (
+  IN EFI_HANDLE  PciHandle
+  )
+{
+  EFI_STATUS           Status;
+  BOOLEAN              LoadOptionRom;
+  EFI_PCI_IO_PROTOCOL  *PciIo;
+  PCI_TYPE00           PciConfHeader;
+
+  LoadOptionRom = FALSE;
+
+  Status = gBS->HandleProtocol (
+      PciHandle,
+      &gEfiPciIoProtocolGuid,
+      (VOID **) &PciIo
+      );
+  if (EFI_ERROR (Status)) {
+      return FALSE;
+  }
+
+  //
+  // Read the PCI Configuration Header from the PCI Device
+  //
+  Status = PciIo->Pci.Read (
+      PciIo,
+      EfiPciIoWidthUint32,
+      0,
+      sizeof (PciConfHeader) / sizeof (UINT32),
+      &PciConfHeader
+      );
+  if (!EFI_ERROR (Status)) {
+      LoadOptionRom = IS_PCI_DISPLAY (&PciConfHeader);
+  }
+
+  gBS->CloseProtocol (
+      PciHandle,
+      &gEfiPciIoProtocolGuid,
+      PciIo,
+      PciHandle
+      );

```

Comment:
We are not doing OpenProtocol, just HandleProtocol, so we shouldn't close the protocol AFAIK @SergiiDmytruk

Reply:
Looks like you're right.  I wasn't sure about it, `PciGetPciRom()` does call `CloseProtocol()` after `HandleProtocol()` and I forgot to check with the spec.

Reply:
I have double checked the spec and it lookes like OpenProtocol is an extended version of HandleProtocol. The difference is that HandleProtocol does not record anywhere if some agent is consuming a protocol. OpenProtocol does record which agent is cnsuming the protocol in a handle database and in such case it CloseProtocol should be required. So we can either:

1. Drop the CloseProtocol, OR
2. Replace HandleProtocol calls with OpenProtocol with attribute EFI_OPEN_PROTOCOL_BY_HANDLE_PROTOCOL

Reply:
OK, I see you went with option 1. Closing discussion